### DISCUSSION TITLE
If the C3 is an evolution, why would you leave a relic of the 1970s ?

### DISCUSSION THREAD

**Discussion by iperov** [c3lang/c3c#1681] [General]
I'm talking about the type system. 
char short int long 
and oh my god, int128 ??  logically it should be long long
where's the logic and consistency?

Do you plan to switch to normal u8 u16 i32 i128 data types ?

**Comment by heavymetalmixer** [comment]
Even though the title is awful I agree with the conect of the post:
Something like i32, i64, etc would be better, mostly not to fall into the same "int is 2 bytes in this platform" trap.

**Reply by lerno** [reply]
I am not sure what ”int is 2 bytes” would be possible given that like D and Java, the sizes are pegged to exact sizes, eg long is always 64 bit, int is always 32 bit

**Comment by lerno** [comment]
Other discussions with the same theme:
https://github.com/c3lang/c3c/discussions/434
https://github.com/c3lang/c3c/issues/1566
https://github.com/c3lang/c3c/issues/1178
https://github.com/c3lang/c3c/issues/660

**Comment by iperov** [comment]
thx. Dead lang on arrival.
Switching to Zig.
